community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/staging/HumanUnderstandableExecutionMonitor.java

```diff
@@ -0,0 +1,466 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.unsafe.impl.batchimport.staging;
+
+import java.io.PrintStream;
+import java.util.TimeZone;
+
+import org.neo4j.graphdb.DependencyResolver;
+import org.neo4j.unsafe.impl.batchimport.CountGroupsStage;
+import org.neo4j.unsafe.impl.batchimport.IdMapperPreparationStage;
+import org.neo4j.unsafe.impl.batchimport.NodeDegreeCountStage;
+import org.neo4j.unsafe.impl.batchimport.NodeStage;
+import org.neo4j.unsafe.impl.batchimport.RelationshipGroupStage;
+import org.neo4j.unsafe.impl.batchimport.RelationshipStage;
+import org.neo4j.unsafe.impl.batchimport.DataStatistics;
+import org.neo4j.unsafe.impl.batchimport.ScanAndCacheGroupsStage;
+import org.neo4j.unsafe.impl.batchimport.SparseNodeFirstRelationshipStage;
+import org.neo4j.unsafe.impl.batchimport.cache.NodeRelationshipCache;
+import org.neo4j.unsafe.impl.batchimport.cache.idmapping.IdMapper;
+import org.neo4j.unsafe.impl.batchimport.input.Input;
+import org.neo4j.unsafe.impl.batchimport.input.Input.Estimates;
+import org.neo4j.unsafe.impl.batchimport.stats.Keys;
+import org.neo4j.unsafe.impl.batchimport.stats.Stat;
+import org.neo4j.unsafe.impl.batchimport.store.BatchingNeoStores;
+
+import static java.lang.Integer.min;
+import static java.lang.Long.max;
+import static java.lang.String.format;
+import static java.lang.System.currentTimeMillis;
+
+import static org.neo4j.helpers.Format.bytes;
+import static org.neo4j.helpers.Format.count;
+import static org.neo4j.helpers.Format.date;
+import static org.neo4j.helpers.Format.duration;
+import static org.neo4j.helpers.collection.Iterables.last;
+import static org.neo4j.unsafe.impl.batchimport.cache.GatheringMemoryStatsVisitor.totalMemoryUsageOf;
+
+/**
+ * Prints progress you can actually understand, with capabilities to on demand print completely incomprehensible
+ * details only understandable to a select few.
+ */
+public class HumanUnderstandableExecutionMonitor implements ExecutionMonitor
```

Comment:
would be nice to have tests for this class to make sure we report progress correctly

Reply:
That will be a bit difficult, but can try.